x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java

```diff
@@ -94,27 +94,33 @@ static String remoteActionDenied(
     ) {
         assert isIndexAction(action);
         String userText = successfulAuthenticationDescription(authentication, authorizationInfo);
+        String remoteClusterText = remoteClusterText(clusterAlias);
+        return actionIsUnauthorizedMessage(action, remoteClusterText, userText)
+            + " because no remote indices privileges apply for the target cluster";
+    }
 
-        return Strings.format(
-            "action [%s] towards remote cluster [%s] is unauthorized for %s"
-                + " because no remote indices privileges apply for the target cluster",
-            action,
-            clusterAlias,
-            userText
-        );
+    private static String remoteClusterText(@Nullable String clusterAlias) {
+        return Strings.format("towards remote cluster%s ", clusterAlias == null ? "" : " [" + clusterAlias + "]");
```

Comment:
I think we could improve the usefulness of the error message (and possibly in an audit context as well) if we include the remote cluster alias from the QC, and pass it along in the headers to the FC. We could store it in authc metadata, and access it here to construct an error message with the alias included. I'd do this in a separate PR; let me know what you think.

Reply:
I think this could be useful especially when query across multiple remote clusters.